File: 2021/0001_twoSum/210621/twoSum.go
func twoSum(nums []int, target int) []int {
    hashTable := map[int]int {}
    for i, n := range nums {
        if p, ok := hashTable[target - n]; ok {
            return []int{p, i}
        }
        hashTable[n] = i
    }
    return nil
}

File: 2021/0001_twoSum/0804/twoSum.ts
function twoSum(nums: number[], target: number): number[] {
    let len = nums.length;
    let map = new Map();
    for (let j: number = 0; j < len; j++) {
        if (map.has(target - nums[j])) {
            return [map.get(target - nums[j]), j];
        }
        map.set(nums[j], j);
    }
    return [];
}
